src/main/java/net/mcreator/preferences/Preferences.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.preferences;
+
+import net.mcreator.io.OS;
+import net.mcreator.preferences.entries.NumberEntry;
+import net.mcreator.preferences.entries.StringEntry;
+import net.mcreator.preferences.entries.PreferenceEntry;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.laf.MCreatorTheme;
+
+import java.awt.*;
+import java.io.File;
+import java.util.Locale;
+
+public class Preferences {
+	// Sections
+	public static final String UI = "ui";
+	public static final String BACKUPS = "backups";
+	public static final String BLOCKLY = "blockly";
+	public static final String IDE = "ide";
+	public static final String GRADLE = "gradle";
+	public static final String BEDROCK = "bedrock";
+	public static final String NOTIFICATIONS = "notifications";
+	public static final String HIDDEN = "hidden";
+
+	// UI
+	public PreferenceEntry<Locale> language;
+	public PreferenceEntry<Color> interfaceAccentColor;
+	public StringEntry backgroundSource;
+	public PreferenceEntry<Boolean> aaText;

```

Comment:
Seeing that boolean type preferences are used a lot, I think you can split such preferences into a separate class. Same about color settings probably, but not the locale type ones (I don't think there may be another language setting at all, so I'd turn the `language` preference into an anonymous subclass and move the locale part of `getComponent()` method to its inheritor in that subclass).